### DISCUSSION TITLE
Use in tauri project?

### DISCUSSION THREAD

**Discussion by panssystem** [sreeise/graph-rs-sdk#391] [Q&A] [answered]
Hi, 

I'm relatively new to Rust, and to this project. I naively set out to try to use it in a desktop project via [Tauri](https://tauri.app/), and discovered that, because Tauri compiles to WASM, it wasn't going to work with the tokio net feature (at least) enabled.

Since this isn't a serious project, only one I'm playing around with as a way to practice with Rust, I wonder if there's anything I could do to help make this possible? Or am I better off using JavaScript libraries to accomplish what I'm looking for?

Thanks!

**Comment by sreeise** [comment, marked as answer]
Hey @panssystem,

I think I understand what your wanting to accomplish but unfortunately this wouldn't be possible because we use a lot of the Tokio library that cannot be included if you want to compile to WASM based on what Tokio says can be used here https://docs.rs/tokio/1.22.0/tokio/index.html#wasm-support. 

Technically, we are using the `full` feature flag of Tokio which just means it has all available features including the ones that cant compile to WASM.
 
Let me know if you have any other questions and id be glad to answer. And congrats on getting started in Rust. I think you will enjoy using it if you stick with it even if you decide to use JavaScript for the project your working on now. Good luck to you.

**Reply by panssystem** [reply]
Thanks that does answer the question. I may play with the unstable option, to see how bad it is, but I'll go a different route for now.
